tests/basic-test.sh

```diff
@@ -181,12 +181,35 @@ cd ${test_tmpdir}
 rm checkout-test2-l -rf
 $OSTREE checkout ${CHECKOUT_H_ARGS} test2 $test_tmpdir/checkout-test2-l
 date > $test_tmpdir/checkout-test2-l/newdatefile.txt
-$OSTREE commit --link-checkout-speedup --consume -b test2 --tree=dir=$test_tmpdir/checkout-test2-l
+$OSTREE commit ${COMMIT_ARGS} --link-checkout-speedup --consume -b test2 --tree=dir=$test_tmpdir/checkout-test2-l
 assert_not_has_dir $test_tmpdir/checkout-test2-l
+$OSTREE fsck
 # Some of the later tests are sensitive to state
 $OSTREE reset test2 test2^
+$OSTREE prune --refs-only
 echo "ok consume (nom nom nom)"
 
+# Test adopt
+cd ${test_tmpdir}
+if grep -q '^mode=bare' repo/mode || is_bare_user_only_repo repo; then
```

Comment:
Did you mean `repo/config`?

This will match `bare-user` too, right? Shouldn't this be `grep -q '^mode=bare$' repo/config`? But really, it seems like we should just always run the test and only do the hardlink check in the non-`bare-user` case?

Reply:
Good idea 💡!  Done ⬇️